File: cpp/recursion/indirect.cpp
#include <iostream>
using namespace std;
void functionB(int n); // as functionB is used before declaration, so compiler to understand we need write prototype for this
void functionA(int n)
{
    if(n > 0) {
        cout << n;
        functionB(n -1 );
    }
}

void functionB(int n) {
    if (n > 1) {
        cout << n;
        functionA(n/2);
    }
}

int main(int argc, char const *argv[])
{
    functionA(20);
    cout << endl;
    return 0;
}

File: java/recursion/Indirect.java
package recursion;

public class Indirect {
    private void functionA(int n) {
        if (n > 0) {
            System.out.print(" " + n);
            this.functionB(n - 1);
        }
    }
    
    private void functionB(int n) {
        if (n > 1) {
            System.out.print(" " + n);
            this.functionA(n/2);
        }
    }
    public static void main(String[] args) {
        (new Indirect()).functionA(20);
        System.out.println();
    }
}